proposals/xxxx_carvel-package/README.md

```diff
@@ -0,0 +1,231 @@
+authors: [ "@benjaminapetersen" ]
+status: "in-review"
+sponsor: [ "@cfryanr", "@joshuatcasey" ]
+approval_date: ""
+---
+
+*Disclaimer*: Proposals are point-in-time designs and decisions.
+Once approved and implemented, they become historical documents.
+If you are reading an old proposal, please be aware that the
+features described herein might have continued to evolve since.
+
+# Carvel Package Management for Pinniped
+
+## Problem Statement
+
+There are a number of tools available to the Kubernetes ecosystem for deploying complex software 
+to a Kubernetes cluster.  The Carvel toolchain provides a set of APIs, Custom Resources and CLI tools
+that can aid a user in the configuration and lifecycle management of software deployed to a cluster.
+We should enhance our deployment options by providing Carvel Packages for the `Suervisor` and `Concierge`
+that may be installed on a cluster configured with Carvel's `kapp-controller` to manage the software 
+on the cluster. 
+
+## How Pinniped Works Today (as of version v0.25.0)
+
+The `./deploy` directory in the root of the Pinniped repository contains a set of `ytt` templates
+that:
+- Are pre-rendered into installable templates listed with each Pinniped relese:
+  - [v0.25.0](https://github.com/vmware-tanzu/pinniped/releases/tag/v0.25.0)
+- Can optionally be customized and rendered by a consumer of the Pinniped project by cloning down
+  the github repository, making changes to the `values.yaml` file and then rendered via `ytt`.
+
+
+## Terminology / Concepts
+
+- `Carvel` is an open-source project that provides tools for managing software build, configuration
+   and deployment on a Kubernetes cluster.  For more information [read the Carvel docs](https://carvel.dev/).
+- `kapp` is a Carvel provided CLI tool for deploying software onto a Kubernetes cluster. See 
+   [the docs](https://carvel.dev/kapp/) for more information.
+- `imgpkg` is a Carvel provided CLI tool that provides a mechanism for collecting configuration and 
+   OCI images into a bundle that can be deployed on a cluster.
+- `kapp-controller` is a server side component managing software in the form of Carvel `App`s delivered
+   via Carvel `Package`s. See [the docs](https://carvel.dev/kapp-controller/) for more information.
+- `PackageRepository` is a custom resource that configures `kapp-controller` with a set of 
+  `Package`s on a cluster.  See [the docs](https://carvel.dev/kapp-controller/docs/v0.47.x/packaging/#package-repository)
+   for more information.
+- `Package` is a custom resource that represents configuration in the form of metadata and OCI images
+   that may be used to deliver software onto a Kuberentes cluster. See [the docs](https://carvel.dev/kapp-controller/docs/v0.47.x/packaging/#package)
+   for more information. 
+- `PackageMetadata` is a custom resource describing attributes for a `Package`.  See [the docs](https://carvel.dev/kapp-controller/docs/v0.47.x/packaging/#package-metadata)
```

Comment:
This looks like the way to specify your custom `values.yaml`, and thus cause your software to be installed. Does that sound correct?

Reply:
As a consumer, to install the software, you'll use a `PackageInstall` and a `Secret`, similar to this:

```yaml
---
apiVersion: packaging.carvel.dev/v1alpha1
kind: PackageInstall
metadata:
    name: "supervisor-package-install"
    namespace: supervisor-install-namespace # should NOT be "supervisor", keep install stuff separate
spec:
  serviceAccountName: "pinniped-package-rbac-supervisor" # will need RBAC
  packageRef:
    refName: "supervisor.pinniped.dev"
    versionSelection:
      constraints: ">0.25.0" #  packages will auto upgrade if they can, up to this constraint
  values:
  - secretRef:
      name: "supervisor-package-install-secret" # secret contains values.yaml values
---
apiVersion: v1
kind: Secret
metadata:
  name: "supervisor-package-install-secret"
    namespace: supervisor-install-namespace # should NOT be "supervisor", keep install stuff separate
# this looks familiar to us!
# this is where the actual values will live.
stringData:
  values.yml: |
    ---
    app_name: pinniped-supervisor
    namespace: supervisor
    api_group_suffix: pinniped.dev
    image_repo: pinniped.local/test/build
    image_tag: 6114CB3E-ACB5-4C67-ABE7-9ACC09B73056
    log_level: debug
    service_https_nodeport_port: 443
    service_https_nodeport_nodeport: 31243
    service_https_clusterip_port: 443

```

Reply:
could we add this to the example `PackageInstall` below?

Reply:
Yup, I can expand on this.